experiment/template/eVOLVER.py

```diff
@@ -482,6 +482,51 @@ def calc_growth_rate(self, vial, gr_start, elapsed_time):
         text_file.write("{0},{1}\n".format(elapsed_time, slope))
         text_file.close()
 
+    def tail_to_np(self, path, window=10, BUFFER_SIZE=512):
+        f = open(path, 'rb')
+        if window == 0:
+            return []
+
+        f.seek(0, os.SEEK_END)
+        remaining_bytes = f.tell()
+        size = window + 1
+        block = -1
+        data = []
+
+        while size > 0 and remaining_bytes > 0:
+            if remaining_bytes - BUFFER_SIZE > 0:
+                # Seek back one whole BUFSIZ
+                f.seek(block * BUFFER_SIZE, 2)
+                # read BUFFER
+                bunch = f.read(BUFFER_SIZE)
+            else:
+                # file too small, start from beginning
+                f.seek(0, 0)
+                # only read what was not read
+                bunch = f.read(remaining_bytes)
+
+            bunch = bunch.decode('utf-8')
+            data.append(bunch)
+            size -= bunch.count('\n')
```

Comment:
Is it possible to break a line in half with this method? Say you read back far enough, but only go halfway through one extra line. Would you then return the broken line in the resulting data set? Or am I missing something?

Reply:
Yes! Sometimes, the first line of the 'bunch' (the last line of the 'bunch' reading from the end) will not be complete. The workaround is to force the algorithm to read always one line extra, and that's where the line `size = window + 1` comes into play.

This way, let's say the bunch has 10 lines but one of them is incomplete: if the user asks for 10 lines, the function will look for 11, reading the next bunch and completed the incomplete line.

I'll add a comment in the code to make it clear.

Reply:
So ideally the broken line is just dropped from the return list, right?

Reply:
Another way to do this is to ready byte by byte, throw those bytes into a buffer, and when you find `\n` you know thats the end of the line. What do you think? I'm fine with this implementation as long as no messed up lines get returned at the end.

Reply:
Yes, a broken line will never go inside the return list.

Maybe something like this would work? Although I am not super confident with the implementation:
https://gist.github.com/dani14-96/0885d2c2ba4663177cf791866f95ce13